metrics/otelgrpc/otelgrpc_test.go

```diff
@@ -0,0 +1,102 @@
+package otelgrpc_test
+
+import (
+	"context"
+	"errors"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/goto/meteor/metrics/otelgrpc"
+	pb "github.com/goto/optimus/protos/gotocompany/optimus/core/v1beta1"
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/otel/attribute"
+	semconv "go.opentelemetry.io/otel/semconv/v1.20.0"
+)
+
+func Test_otelGRPCMonitor_Record(t *testing.T) {
+	mt := otelgrpc.NewOtelGRPCMonitor("localhost:1001")
+	assert.NotNil(t, mt)
+	initialAttr := mt.GetAttributes()
+
+	uc := mt.UnaryClientInterceptor()
+	assert.NotNil(t, uc)
+	assert.Equal(t, initialAttr, mt.GetAttributes())
+
+	sc := mt.StreamClientInterceptor()
+	assert.NotNil(t, sc)
+	assert.Equal(t, initialAttr, mt.GetAttributes())
+
+	mt.RecordUnary(context.Background(), otelgrpc.UnaryParams{
+		Start:  time.Now(),
+		Method: "/service.gojek.com/MethodName",
+		Req:    nil,
+		Res:    nil,
+		Err:    nil,
+	})
+	assert.Equal(t, initialAttr, mt.GetAttributes())
+
+	mt.RecordUnary(context.Background(), otelgrpc.UnaryParams{
+		Start:  time.Now(),
+		Method: "",
+		Req:    &pb.ListProjectsRequest{},
+		Res:    nil,
+		Err:    nil,
+	})
+	assert.Equal(t, initialAttr, mt.GetAttributes())
+
+	mt.RecordStream(context.Background(), time.Now(), "", nil)
+	assert.Equal(t, initialAttr, mt.GetAttributes())
+
+	mt.RecordStream(context.Background(), time.Now(), "/service.gojek.com/MethodName", errors.New("dummy error"))
+	assert.Equal(t, initialAttr, mt.GetAttributes())
+}
+
+func Test_parseFullMethod(t *testing.T) {
+	type args struct {
+		fullMethod string
+	}
+	tests := []struct {
```

Comment:
Should we add the failure cases as well?

Reply:
added